src/arith/num_big.hpp

```
@@ -110,146 +266,83 @@ class Number final {
-    Number zero_extend_impl() const {
-        using namespace boost::multiprecision;
-        static const cpp_int value_mask = (cpp_int(1) << width) - 1;
-        return cpp_int(_n & value_mask);
+        constexpr auto mask = static_cast<UInt128>(std::numeric_limits<U>::max());
+        return static_cast<T>(static_cast<U>(static_cast<UInt128>(_n) & mask));
     }
 
-    // Allow truncating to signed int as needed for finite width operations.
-    // Unlike casting, sign_extend will not throw a prevail error if the number doesn't fit.
     [[nodiscard]]
     Number sign_extend(const int width) const {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)